src/cpp/src/text2image/stable_diffusion_xl_pipeline.hpp

```diff
@@ -0,0 +1,345 @@
+        const std::string model_index_path = root_dir + "/model_index.json";
+        std::ifstream file(model_index_path);
+        OPENVINO_ASSERT(file.is_open(), "Failed to open ", model_index_path);
+
+        nlohmann::json data = nlohmann::json::parse(file);
+        using utils::read_json_param;
+
+        set_scheduler(Scheduler::from_config(root_dir + "/scheduler/scheduler_config.json"));
+
+        const std::string text_encoder = data["text_encoder"][1].get<std::string>();
+        if (text_encoder == "CLIPTextModel") {
+            m_clip_text_encoder = std::make_shared<CLIPTextModel>(root_dir + "/text_encoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        const std::string text_encoder_2 = data["text_encoder_2"][1].get<std::string>();
+        if (text_encoder_2 == "CLIPTextModelWithProjection") {
+            m_clip_text_encoder_with_projection = std::make_shared<CLIPTextModelWithProjection>(root_dir + "/text_encoder_2", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        const std::string unet = data["unet"][1].get<std::string>();
+        if (unet == "UNet2DConditionModel") {
+            m_unet = std::make_shared<UNet2DConditionModel>(root_dir + "/unet", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", unet, "' UNet type");
+        }
+
+        const std::string vae = data["vae"][1].get<std::string>();
+        if (vae == "AutoencoderKL") {
+            m_vae_decoder = std::make_shared<AutoencoderKL>(root_dir + "/vae_decoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", vae, "' VAE decoder type");
+        }
+
+        // initialize generation config
+        initialize_generation_config(data["_class_name"].get<std::string>());
+    }
+
+    StableDiffusionXLPipeline(
+        const CLIPTextModel& clip_text_model,
+        const CLIPTextModelWithProjection& clip_text_model_with_projection,
+        const UNet2DConditionModel& unet,
+        const AutoencoderKL& vae_decoder)
+        : m_clip_text_encoder(std::make_shared<CLIPTextModel>(clip_text_model)),
+          m_clip_text_encoder_with_projection(std::make_shared<CLIPTextModelWithProjection>(clip_text_model_with_projection)),
+          m_unet(std::make_shared<UNet2DConditionModel>(unet)),
+          m_vae_decoder(std::make_shared<AutoencoderKL>(vae_decoder)) { }
+
+    void reshape(const int num_images_per_prompt, const int height, const int width, const float guidance_scale) override {
+        check_inputs(height, width);
+
+        const size_t batch_size_multiplier = do_classifier_free_guidance(guidance_scale) ? 2 : 1;  // Unet accepts 2x batch in case of CFG
+        m_clip_text_encoder->reshape(batch_size_multiplier);
+        // TODO: m_unet reshape
```

Comment:
do we need this TODO? Looks like Unet is reshaped

I suppose we need to reshape `m_clip_text_encoder_with_projection`